lnwallet/channel.go

```diff
@@ -209,6 +209,35 @@ type PaymentDescriptor struct {
 	// Settle.
 	ParentIndex uint64
 
+	// SourceRef points to an Add update in a forwarding package owned by
+	// this channel.
+	//
+	// NOTE: This field will only be populated if EntryType is Fail or
+	// Settle.
+	SourceRef *channeldb.AddRef
+
+	// DestRef points to a Fail/Settle update in another link's forwarding
+	// package.
+	//
+	// NOTE: This field will only be populated if EntryType is Fail or
+	// Settle, and the forwarded Add successfully included in an outgoing
+	// link's commitment txn.
+	DestRef *channeldb.SettleFailRef
+
+	// OpenCircuitRef references the incoming Chan/HTLC ID of an HTLC packet
+	// delivered by the switch.
+	//
+	// NOTE: This field is only populated for payment descriptors in the
```

Comment:
When would this not be the case?

Reply:
Won't be set if the payment descriptor is for the remote log, or if this is a fail/settle. Will update to be more clear